domains/transfer-agreements/TransferAgreements.API/API/Program.cs

```diff
@@ -33,6 +37,40 @@
 builder.Services.AddOptions<DatabaseOptions>().BindConfiguration(DatabaseOptions.Prefix).ValidateDataAnnotations().ValidateOnStart();
 builder.Services.AddDbContext<ApplicationDbContext>((sp, options) => options.UseNpgsql(sp.GetRequiredService<IOptions<DatabaseOptions>>().Value.ToConnectionString()));
 
+Audit.Core.Configuration.Setup()
+    .UseEntityFramework(ef => ef
+        .AuditTypeExplicitMapper(config => config
+            .Map<TransferAgreement, TransferAgreementHistoryEntry>((evt, eventEntry, historyEntity) =>
+            {
+                if (eventEntry.Entity != null && eventEntry.Entity.GetType() == typeof(TransferAgreementHistoryEntry))
+                {
+                    return false;
+                }
+
+                var actorId = evt.CustomFields.ContainsKey("ActorId") ? evt.CustomFields["ActorId"].ToString() : null;
+                var actorName = evt.CustomFields.ContainsKey("ActorName") ? evt.CustomFields["ActorName"].ToString() : null;
+
+                historyEntity.Id = Guid.NewGuid();
+                historyEntity.AuditDate = DateTime.UtcNow;
+                historyEntity.AuditAction = eventEntry.Action;
+                historyEntity.ActorId = actorId;
+                historyEntity.ActorName = actorName;
+
+                switch (eventEntry.Action)
```

Comment:
Is this also called on "Select"-statements on TransferAgreement-table/DbSet? It is not something that we want to be saved.

Reply:
In the Audit.NET configuration, there is no explicit logic to disable auditing for Select operations or enable it only for Insert and Update. It's just the default behavior of the library.

Reply:
So to answer your question. No it is not enabled for select statements. Only operations, that change the state of the entity. In PostgreSQL language, that means Insert, Update, Delete operations ONLY.